app/workers/enqueue_subject_queue_worker.rb

```diff
@@ -13,64 +13,18 @@ class EnqueueSubjectQueueWorker
     }
   })
 
-  attr_reader :queue, :workflow, :user, :subject_set_id, :limit, :selection_strategy
-
   def perform(queue_id, limit=SubjectQueue::DEFAULT_LENGTH, strategy_override=nil)
-    @queue = SubjectQueue.find(queue_id)
-    @workflow = queue.workflow
-    @user = queue.user
-    @subject_set_id = queue.subject_set_id
-    @limit = limit
-    @selection_strategy = strategy(strategy_override)
-
-    sms_ids = strategy_sms_ids
-    unless sms_ids.empty?
-      user_seens = user ? UserSeenSubject.find_by(user: user, workflow: workflow) : nil
-      unseen_ids = Subjects::SeenRemover.new(user_seens, sms_ids).unseen_ids
-      queue.enqueue_update(unseen_ids)
-    end
+    queue = SubjectQueue.find(queue_id)
+    selector = Subjects::StrategySelector.new(
```

Comment:
I didn't like the name either...will come up with something enterprisey for you!